src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtil.java

```diff
@@ -123,20 +123,8 @@ public static boolean matchesFileExtension(File file, String... fileExtensions)
             result = true;
         }
         else {
-            // normalize extensions so all of them have a leading dot
-            final String[] withDotExtensions = new String[fileExtensions.length];
-            for (int i = 0; i < fileExtensions.length; i++) {
-                final String extension = fileExtensions[i];
-                if (extension.startsWith(EXTENSION_SEPARATOR)) {
-                    withDotExtensions[i] = extension;
-                }
-                else {
-                    withDotExtensions[i] = EXTENSION_SEPARATOR + extension;
-                }
-            }
-
             final String fileName = file.getName();
```

Comment:
@romani do we really required to do this at the end `fileName.endsWith(fileExtension)` we are checking only wether file names end with fileExtension.

Reply:
I am surprised on this code 
The only place where we do mentioning `.` in extension name is at https://checkstyle.org/config.html#TreeWalker_Properties

All other places are without dot.

Reply:
Extension is not a standard , it is just agreement that file name ends with some characters.

We should not parse java file that is named as `Test,java` so looks like we missed test on names without dots as separator of name and extension.

Reply:
Test added